src/providers/push/fcm/fcm.provider.ts

```diff
@@ -0,0 +1,128 @@
+import {inject, Provider} from '@loopback/core';
+import {HttpErrors} from '@loopback/rest';
+import * as admin from 'firebase-admin';
+import {app as FirebaseAdmin} from 'firebase-admin';
+import {ConditionMessage, TopicMessage} from 'firebase-admin/lib/messaging';
+import {FcmBindings} from './keys';
+import {FcmConfig, FcmMessage, FcmNotification} from './types';
+
+export class FcmProvider implements Provider<FcmNotification> {
+  constructor(
+    @inject(FcmBindings.Config, {
+      optional: true,
+    })
+    private readonly fcmConfig?: FcmConfig,
+  ) {
+    if (this.fcmConfig) {
+      this.fcmService = admin.initializeApp({
+        credential: admin.credential.cert(this.fcmConfig.serviceAccountPath),
+        databaseURL: this.fcmConfig.dbUrl,
+      });
+    } else {
+      throw new HttpErrors.PreconditionFailed('Firebase Config missing !');
+    }
+  }
+
+  fcmService: FirebaseAdmin.App;
+
+  value() {
+    return {
+      publish: async (message: FcmMessage) => {
+        if (
+          message.receiver.to.length === 0 &&
+          !message.options.topic &&
+          !message.options.condition
```

Comment:
why is condition mandatory ?

Reply:
we can't send push without any one of these i.e tokens, topic or condition